pkg/ruler/api.go

```diff
@@ -268,3 +284,241 @@ func (r *Ruler) alerts(w http.ResponseWriter, req *http.Request) {
 		level.Error(logger).Log("msg", "error writing response", "bytesWritten", n, "err", err)
 	}
 }
+
+var (
+	// ErrNoNamespace signals that no namespace was specified in the request
+	ErrNoNamespace = errors.New("a namespace must be provided in the request")
+	// ErrNoGroupName signals a group name url parameter was not found
+	ErrNoGroupName = errors.New("a matching group name must be provided in the request")
+	// ErrNoRuleGroups signals the rule group requested does not exist
+	ErrNoRuleGroups = errors.New("no rule groups found")
+	// ErrBadRuleGroup is returned when the provided rule group can not be unmarshalled
+	ErrBadRuleGroup = errors.New("unable to decoded rule group")
+)
+
+// ValidateRuleGroup validates a rulegroup
+func ValidateRuleGroup(g rulefmt.RuleGroup) []error {
+	var errs []error
+	for i, r := range g.Rules {
+		for _, err := range r.Validate() {
+			var ruleName string
+			if r.Alert != "" {
+				ruleName = r.Alert
+			} else {
+				ruleName = r.Record
+			}
+			errs = append(errs, &rulefmt.Error{
+				Group:    g.Name,
+				Rule:     i,
+				RuleName: ruleName,
+				Err:      err,
+			})
+		}
+	}
+
+	return errs
+}
+
+func marshalAndSend(formatted interface{}, w http.ResponseWriter, logger log.Logger) {
```

Comment:
nit: I wouldn't call it `formatted` here, perhaps just output, or result.

Reply:
👍